urbanairship-preference-center-react-native/ios/AirshipPreferenceCenterModule.m

```diff
@@ -45,18 +45,22 @@ @implementation AirshipPreferenceCenterModule
                     NSArray* items = section.items;
                     if (items) {
                         NSMutableArray *itemArray = [NSMutableArray array];
-                        for (id<UAPreferenceItem> item in items) {
-                            NSMutableDictionary *itemDictionary = [NSMutableDictionary dictionary];
-                            [itemDictionary setValue:item.identifier forKey:@"id"];
-                            
-                            UAPreferenceCommonDisplay* itemCommonDisplay = item.display;
-                            if (itemCommonDisplay) {
-                                NSMutableDictionary *itemDisplayDictionary = [NSMutableDictionary dictionary];
-                                [itemDisplayDictionary setValue:itemCommonDisplay.title forKey:@"name"];
-                                [itemDisplayDictionary setValue:itemCommonDisplay.subtitle forKey:@"description"];
-                                [itemDictionary setValue:itemDisplayDictionary forKey:@"display"];
+                        for (id item in items) {
+                            if ([item isKindOfClass:[UAPreferenceChannelSubscriptionItem class]]) {
```

Comment:
I feel like you should have add only the subscriptionId if it's a UAPreferenceChannelSubscriptionItem. The other fields are present in UAPreferenceItem.